### ISSUE TITLE
Error in certificate_info command when trying to parse OCSP response

### ISSUE THREAD

**Issue by faheel** [nabla-c0d3/sslyze#471]
**Describe the bug**

I'm getting an `IndexError` in the `certificate_info` command when trying to parse OCSP response. Maybe this occurs for domains that do not have any certificate? This is similar to the closed issue #254.

**To Reproduce**

1. Install SSLyze using `pip`
2. Run the following command:
   ```bash
   python -m sslyze --regular e-saksham.nic.in
   ```
3. See the error:
```
 CHECKING HOST(S) AVAILABILITY
 -----------------------------

   e-saksham.nic.in:443                       => 164.100.229.145 




 SCAN RESULTS FOR E-SAKSHAM.NIC.IN:443 - 164.100.229.145
 -------------------------------------------------------

 * TLS 1.3 Cipher Suites:
     Attempted to connect using 5 cipher suites; the server rejected all cipher suites.

 * TLS 1.0 Cipher Suites:
     Attempted to connect using 80 cipher suites.

     The server accepted the following 2 cipher suites:
        TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA                256       ECDH: prime256v1 (256 bits)
        TLS_DHE_RSA_WITH_AES_256_CBC_SHA                  256       DH (2048 bits) 

     The group of cipher suites supported by the server has the following properties:
       Forward Secrecy                    OK - Supported
       Legacy RC4 Algorithm               OK - Not Supported

     The server is configured to prefer the following cipher suite:
        TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA                256       ECDH: prime256v1 (256 bits)


 * TLS 1.2 Session Resumption Support:
      With Session IDs: OK - Supported (5 successful resumptions out of 5 attempts).
      With TLS Tickets: NOT SUPPORTED - Server did not return a TLS ticket.

 * ROBOT Attack:
                                          OK - Not vulnerable, RSA cipher suites not supported.

 * TLS 1.1 Cipher Suites:
     Attempted to connect using 80 cipher suites.

     The server accepted the following 2 cipher suites:
        TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA                256       ECDH: prime256v1 (256 bits)
        TLS_DHE_RSA_WITH_AES_256_CBC_SHA                  256       DH (2048 bits) 

     The group of cipher suites supported by the server has the following properties:
       Forward Secrecy                    OK - Supported
       Legacy RC4 Algorithm               OK - Not Supported

     The server is configured to prefer the following cipher suite:
        TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA                256       ECDH: prime256v1 (256 bits)


 * OpenSSL Heartbleed:
                                          OK - Not vulnerable to Heartbleed

 * SSL 3.0 Cipher Suites:
     Attempted to connect using 80 cipher suites; the server rejected all cipher suites.

 * Elliptic Curve Key Exchange:
       Supported curves:                  X25519, prime256v1, secp384r1, secp521r1
       Rejected curves:                   secp224k1, secp160k1, sect239k1, secp224r1, X448, secp160r1, sect283k1, secp192k1, secp256k1, secp160r2, sect283r1, sect163r2, prime192v1, sect409k1, sect233r1, sect163k1, sect193r1, sect409r1, sect193r2, sect571k1, sect233k1, sect163r1, sect571r1

 * SSL 2.0 Cipher Suites:
     Attempted to connect using 7 cipher suites; the server rejected all cipher suites.

 * OpenSSL CCS Injection:
                                          OK - Not vulnerable to OpenSSL CCS injection

 * Downgrade Attacks:
       TLS_FALLBACK_SCSV:                 OK - Supported

 * TLS 1.2 Cipher Suites:
     Attempted to connect using 158 cipher suites.

     The server accepted the following 10 cipher suites:
        TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384             256       ECDH: prime256v1 (256 bits)
        TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA384             256       ECDH: prime256v1 (256 bits)
        TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA                256       ECDH: prime256v1 (256 bits)
        TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256             128       ECDH: prime256v1 (256 bits)
        TLS_DHE_RSA_WITH_AES_256_GCM_SHA384               256       DH (2048 bits) 
        TLS_DHE_RSA_WITH_AES_256_CCM_8                    256       DH (2048 bits) 
        TLS_DHE_RSA_WITH_AES_256_CCM                      256       DH (2048 bits) 
        TLS_DHE_RSA_WITH_AES_256_CBC_SHA256               256       DH (2048 bits) 
        TLS_DHE_RSA_WITH_AES_256_CBC_SHA                  256       DH (2048 bits) 
        TLS_DHE_RSA_WITH_AES_128_GCM_SHA256               128       DH (2048 bits) 

     The group of cipher suites supported by the server has the following properties:
       Forward Secrecy                    OK - Supported
       Legacy RC4 Algorithm               OK - Not Supported

     The server is configured to prefer the following cipher suite:
        TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384             256       ECDH: prime256v1 (256 bits)


 * Deflate Compression:
                                          OK - Compression disabled

 * Session Renegotiation:
       Client-initiated Renegotiation:    OK - Rejected
       Secure Renegotiation:              OK - Supported

 * Error when running --certinfo:
       You can open an issue at https://github.com/nabla-c0d3/sslyze/issues with the following information:

       * Server: e-saksham.nic.in:443 - 164.100.229.145
       * Scan command: certificate_info

       Traceback (most recent call last):
         File "/home/faheel/.local/lib/python3.8/site-packages/sslyze/scanner.py", line 264, in get_results
    result = implementation_cls.result_for_completed_scan_jobs(
         File "/home/faheel/.local/lib/python3.8/site-packages/sslyze/plugins/certificate_info/implementation.py", line 100, in result_for_completed_scan_jobs
    received_chain_as_pem, ocsp_response, custom_ca_file = completed_job.result()
         File "/home/faheel/.asdf/installs/python/3.8.2/lib/python3.8/concurrent/futures/_base.py", line 432, in result
    return self.__get_result()
         File "/home/faheel/.asdf/installs/python/3.8.2/lib/python3.8/concurrent/futures/_base.py", line 388, in __get_result
    raise self._exception
         File "/home/faheel/.asdf/installs/python/3.8.2/lib/python3.8/concurrent/futures/thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
         File "/home/faheel/.local/lib/python3.8/site-packages/sslyze/plugins/certificate_info/_get_cert_chain.py", line 27, in get_certificate_chain
    ocsp_response = ssl_connection.ssl_client.get_tlsext_status_ocsp_resp()
         File "/home/faheel/.local/lib/python3.8/site-packages/nassl/ssl_client.py", line 371, in get_tlsext_status_ocsp_resp
    return OcspResponse.from_openssl(ocsp_response)
         File "/home/faheel/.local/lib/python3.8/site-packages/nassl/ocsp_response.py", line 102, in from_openssl
    version=int(_get_value_from_text_output_no_p("Version:", response_text)),
         File "/home/faheel/.local/lib/python3.8/site-packages/nassl/ocsp_response.py", line 140, in _get_value_from_text_output_no_p
    value = _get_value_from_text_output(key, text_output)
         File "/home/faheel/.local/lib/python3.8/site-packages/nassl/ocsp_response.py", line 136, in _get_value_from_text_output
    return value[1].split("\n")[0].strip()
       IndexError: list index out of range


 SCAN COMPLETED IN 4.58 S
 ------------------------
```

**Expected behavior**
The `certificate_info` scan command should quietly handle this case (or maybe display a warning) and not throw any exception.

**Python environment:**
 - OS: Ubuntu 20.04
 - Python version: 3.8
 - sslyze v3.1.0
 - nassl v3.1.0

**Comment by gattjoe** [comment]
Hi faheel,  I *think* the error here is that the OCSP response back from the CA for your cert is "unauthorized".  See https://tools.ietf.org/html/rfc6960.

This *could* mean that the OCSP responder is not authorized to give a response for your cert (possibly because it is expired).

I'll take a look at the code later and see if it can handle an "unauthorized" response back from an OCSP responder.

**Comment by gattjoe** [comment]
@nabla-c0d3 I think the issue is actually in nassl in ocsp_response.py.  On line 135, the code `value = text_output.split(key)` is called and in this case, the key is "Version:".  A good OCSP response from digicert.com looks like this:

_"OCSP Response Data:\n    OCSP Response Status: successful (0x0)\n    Response Type: Basic OCSP Response\n    Version: 1 (0x0)\n    Responder Id: 3DD350A5D6A0ADEEF34A600A65D321D4F8F8D60F\n    Produced At: Jan  4 07:27:02 2021 GMT\n    Responses:\n    Certificate ID:\n      Hash Algorithm: sha1\n...(truncated)._

The OCSP response for this cert is "Try Later":

_"OCSP Response Data:\n    OCSP Response Status: trylater (0x3)\n"_

So the code will split the string by the key so you can parse out the OCSP response. Line 136 is what is causing the issue.  I'd submit a PR, but I am not familiar enough with the code to know if checking the length or some other bounds check will create an edge case somewhere.

cc @faheel

**Comment by nabla-c0d3** [comment]
Hello and thanks for the bug report. @gattjoe Correct, there is a bug in the text parsing logic in nassl; it will be fixed in the next release.

**Comment by nabla-c0d3** [comment]
The fix was released as part of [v4.0.0](https://github.com/nabla-c0d3/sslyze/releases/tag/4.0.0).
